### PR TITLE
Add Option to Disable More Feedback Requests

### PR DESCRIPTION
**Pull request by MaisiKoleni**
### Checklist
- [x] The database changes got approved. --- **Do not test this PR on the test servers until this is checked!**
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
Some courses need to be able to adjust the assessment after a complaint if it is wrong because exam-rules apply, practical courses such as PGdP have this requirement. Student using the more feedback function only make everything more complicated, 
because the score cannot be modified by tutors. We want to disable more feedback requests for the course and only allow (unlimited) complaints because we more or less have to.

### Description
- **Database Change: added `max_request_more_feedback_time_days` column**
  It is not nullable and a primitive int.
- Added a corresponding attribute to the `Course` class and adjusted some logic
- Changed course and complaint validation (on create/update)
- Added the attribute on the client side as well
- Adjust client forms and some complaint-setting-based conditions

#### Open Questions
- [ ] Find out why team complaints do not work at all (even on develop, or am I doing something wrong?)
  Seems to be decided by `this.isCurrentUserSubmissionAuthor = this.participation.student.id === user.id;`, is that a problem?
- [ ] The same goes for more feedback requests for teams, those don't seem to exist as well.

### Steps for Testing
0. Review the code, especially look if I missed something that needs to be changed as well
1. Test that complaints with various course settings work:
    1. Request more feedback enabled <-> disabled
    4. Complaints enabled <-> disabled.
2. Adjust the course setting and then see, if the correct options are available for an _already graded submission in this course_.
    "Request More Feedback" should only be available if the option is checked, and "Complain" only when complaint tokens are available.
3. Submit a Complaint / More Feedback Request and check that it is accepted by the server (e.g. check on the Instructor Dashboard)

### Test Coverage
_more or less unchanged_

### Screenshots
- **Options**
  image
- **Students can only submit complaints and not request more feedback if disabled**
  image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java [hunk 1]
```diff
@@ -224,22 +226,44 @@ public Integer getMaxComplaintsPerParticipant(Course course, Participant partici
...
+        }
+
+        if (!isTimeValid) {
+            StringBuilder message = new StringBuilder("You cannot ");
+            message.append(type == ComplaintType.COMPLAINT ? "submit a complaint" : "request more feedback");
+            message.append(" for a result that is older than ");
+            message.append(switch (maxDays) {
+                case 1 -> "one day";
+                case 7 -> "one week";
+                default -> maxDays + (maxDays % 7 == 0 ? " weeks" : " days");
```
Comment (kloessst) [inline src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java:255]: If maxDays is 14 this will result in "14 weeks" right?

**Review by kloessst — CHANGES_REQUESTED** [review]
Code looks good just a small note

File: src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java (hunk 1 above)
Comment (MaisiKoleni) [reply src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java:255]: Yes, you are right, good catch. I will fix this.

**Review by airelawaleria — APPROVED** [review]
Works fine! Tested in the testing session. Undefined bahaviour when some complaints were handled, and then the complaint feature gets disabled.

**Review by EneaGore — APPROVED** [review]
On testing session all worked as intended. Undefined behaviour when complaints have been submitted and then are disabled.

**Review by tiffca — APPROVED** [review]
Tested on ts1. Works as expected!

Note: The student can see the complaint when complaint is disabled and feedback is enabled. The student cannot see the complaint when complaint and feedback are disabled. 
TBD: What should the behavior be when both complaints and feedback requests are disabled? In this scenario, the previously submitted complaints are no longer shown to the student. Is this as expected? Or students should still be able to see the result of their previous complaints even when the complaint option is disabled.

**Review by philgabr — DISMISSED** [review]
Basically everything works fine. Tested in the testing session.
It could be discussed if students should still be able to see former complaints when complaining is disabled afterwards.

**Review by JohannesWeiss — DISMISSED** [review]
The new function did work as intended. Tested independent from test session and verified with them.

**Review by kloessst — APPROVED** [review]
Code looks good to me.

**Review by airelawaleria — APPROVED** [review]
Code looks fine, works as intended

**Review by EneaGore — APPROVED** [review]
Looks good.

**Review by philgabr — APPROVED** [review]
looks reasonable

**Review by airelawaleria — APPROVED** [review]
Works fine! Tested in the testing session. And after the changes with @simonzilker

**Review by simonzilker — APPROVED** [review]
Everything worked as described!

**Review by EneaGore — APPROVED** [review]
Code looks fine.

**Pull request merged**